### PR TITLE
feat: select and multiselect label value

### PR DESCRIPTION
**Pull request by sbalaji1192**
## Description
- Select and multi-select widgets now have two new properties under the data section label and value.
- The existing options have been renamed to Source data.
- Users can set the label and value of the options through these new properties.
- We have written migrations to make this work for existing select widgets.

#### PR fixes following issue(s)
Fixes https://github.com/appsmithorg/appsmith/issues/24022

>
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [x] Manual
- [x] Jest
- [x] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by chandannkumar** [comment]
Observation:
- Show error message on when there is no label/value selected from dropdown
- Show duplication error when source data have duplicate values

**Comment by chandannkumar** [comment]
Tested this PR and LGTM
- Tested Select & Multiselect widget
- Tested migration scenarios
- Tested on modal & containers

File: app/client/src/widgets/SelectWidget/widget/propertyUtils.ts [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+    }
+  }
+  return {
+    isValid,
+    parsed,
+    messages: [message],
+  };
+}
+
+export function labelValueKeyOptions(widget: WidgetProps) {
```
Comment (keyurparalkar) [inline app/client/src/widgets/SelectWidget/widget/propertyUtils.ts:107]: Can we rename this to: `generateLabelValueOptionsFromSourceDataKeys`?

File: app/client/src/widgets/MultiSelectWidgetV2/widget/propertyUtils.ts [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+    }
+  }
+  return {
+    isValid,
+    parsed,
+    messages: [message],
+  };
+}
+
+export function labelValueKeyOptions(widget: WidgetProps) {
```
Comment (keyurparalkar) [inline app/client/src/widgets/MultiSelectWidgetV2/widget/propertyUtils.ts:182]: Can we rename this to: generateLabelValueOptionsFromSourceDataKeys?

**Comment by chandannkumar** [comment]
LGTM

**Comment by chandannkumar** [comment]
LGTM

**Review by keyurparalkar — APPROVED** [review]

**Pull request merged**